server/UMD_server/UMD_dataset/views.py

```diff
@@ -19,8 +19,14 @@
 
 def mapUserIds(users):
     userMap = {}
+    conflictMap = {}
     for item in users:
-        uid = int(str(item["_id"]), 16)
+        base_uid = int(str(item["_id"]), 16)
+        # lets grab the last 5 digits
+        uid = int(str(base_uid)[-5:])
+        while uid in conflictMap.keys():  # ensure a unique ID
+            uid = uid + 1
+        conflictMap[uid] = True
```

Comment:
Is it possible that this algorithm would give different IDs for the same user under different sets of users?

Reply:
Yes, but I'm not sure if it matters the way the data is used.  The userId I believe only needs to be unique per the export of the data.  If there were conflicting end values for the girderId of users and one of the users who was conflicting was removed it could give a different userId.  Although the userID is meant to associate within a single sample dataset export for the NIST tool.  So it's important that there are no two same Ids within an export, but if items change between the exporting there is still the main userMap for that export which can get back to a true UserId.

I.E. John has a UID of 1000 and Jane also has a base UID of 1000 so Jane's UID is changed to 1001.  Then if you were to remove all annotations by John then Jane could have a UID of 1000 instead of 1001.  Either way, when the data is exported the UserMap.tab will have the direct mapping back to their girderId/login/username.  The exporting of the data is done as a whole and not incrementally (clicking export creates a complete package of the data).

I wanted to base the root number off of something that must be unique (girderId) but still wanted to account for the issue in which those numbers weren't unique because of the digits and didn't have an ordering system.